### PR TITLE
20221118-minor-fixes

### PR DESCRIPTION
**Pull request by douzzer**
`configure.ac`: mutually harmonize `--enable-all` and `--enable-all-crypto`: add `enable-siphash` to `enable-all`, and add Brainpool to `enable-all-crypto`.

`src/internal.c`: in `VerifyServerSuite()`, narrow condition and fix return value in error check added in 647ce794dd.

`wolfcrypt/src/siphash.c`: fix `bugprone-macro-parentheses`, and in `wc_SipHash()`, fix `clang-diagnostic-overlength-strings` around inline asm.

tested with `wolfssl-multi-test.sh ... super-quick-check benchmark-wolfcrypt-intelasm-all benchmark-wolfcrypt-noasm-all`

### CODE REVIEW

**Review by dgarske — APPROVED** [review]

File: wolfcrypt/src/siphash.c [hunk 1]
```diff
@@ -465,6 +465,12 @@ int wc_SipHash(const unsigned char* key, const unsigned char* in, word32 inSz,
         "orq    %%r12, %%r13\n\t"
         "L_siphash_n6:\n\t"
 
+    : [in] "+r" (in), [inSz] "+r" (inSz)
```
Comment (SparkiDev) [inline wolfcrypt/src/siphash.c:468]: Why was this needed?
It is dangerous to split assembly code in the middle like this as the used registers aren't necessarily still the same values.
